packaging/centos/datadog-agent.init

```diff
@@ -51,11 +62,35 @@ check_status() {
         s=`$SUPERVISORCTL_PATH -c $SUPERVISOR_CONF status`
 
         # number of RUNNING supervisord programs (ignoring dogstatsd, jmxfetch and go-metro)
-	p=`echo "$s" | grep -Ev 'dogstatsd|jmxfetch|go-metro' | grep -c RUNNING`
+        p=`echo "$s" | grep -Ev $DD_OPT_PROC_REGEX | grep -c RUNNING`
+
+        # Number of non-failed OPTIONAL supervisord programs (dogstatsd, jmxfetch and go-metro)
+        p_o=`echo "$s" | grep -E $DD_OPT_PROC_REGEX | grep -c FATAL`
```

Comment:
Good catch! Must've inadvertedly removed it in vim, because this was a paste from the `debian` version!